pkg/loaders/azure/models/job.go

```diff
@@ -0,0 +1,168 @@
+}
+
+type CIJob struct {
+	Job           string       `yaml:"job,omitempty"`
+	Strategy      *JobStrategy `yaml:"strategy,omitempty"`
+	BaseJob       `yaml:",inline"`
+	FileReference *models.FileReference
+}
+
+type DeploymentEnvironment struct {
+	Name          string `yaml:"name,omitempty"`
+	ResourceName  string `yaml:"resourceName,omitempty"`
+	ResourceId    string `yaml:"resourceId,omitempty"`
+	ResourceType  string `yaml:"resourceType,omitempty"`
+	Tags          string `yaml:"tags,omitempty"`
+	FileReference *models.FileReference
+}
+
+func (de *DeploymentEnvironment) UnmarshalYAML(node *yaml.Node) error {
+	de.FileReference = loadersUtils.GetFileReference(node)
+	if node.Tag == consts.StringTag {
+		de.Name = node.Value
+		return nil
+	}
+
+	return node.Decode(&de)
+}
+
+type DeploymentJob struct {
+	Deployment    string                 `yaml:"deployment,omitempty"`
+	Strategy      *DeploymentStrategy    `yaml:"strategy,omitempty"`
+	Environment   *DeploymentEnvironment `yaml:"environment,omitempty"`
+	BaseJob       `yaml:",inline"`
+	FileReference *models.FileReference
+}
+
+type TemplateJob struct {
+	Template      `yaml:",inline"`
+	FileReference *models.FileReference
+}
+
+type Jobs struct {
+	CIJobs         *[]CIJob
+	DeploymentJobs *[]DeploymentJob
+	TemplateJobs   *[]TemplateJob
+	FileReference  *models.FileReference
+}
+
+func (j *Jobs) UnmarshalYAML(node *yaml.Node) error {
+	var ciJobs []CIJob
+	var deploymentJobs []DeploymentJob
+	var templateJobs []TemplateJob
+
+	for _, jobNode := range node.Content {
+		switch getJobType(jobNode) {
+		case CIJobType:
+			job, err := parseCIJob(jobNode)
+			if err != nil {
+				return err
+			}
+			ciJobs = append(ciJobs, job)
+		case DeploymentJobType:
+			job, err := parseDeploymentJob(jobNode)
+			if err != nil {
+				return err
+			}
+			deploymentJobs = append(deploymentJobs, job)
+		case TemplateJobType:
+			job, err := parseTemplateJob(jobNode)
+			if err != nil {
+				return err
+			}
+			templateJobs = append(templateJobs, job)
+		}
+	}
+
+	*j = Jobs{
+		CIJobs:         &ciJobs,
+		DeploymentJobs: &deploymentJobs,
+		TemplateJobs:   &templateJobs,
+		FileReference:  loadersUtils.GetFileReference(node),
+	}
+	return nil
+}
+
+func parseCIJob(node *yaml.Node) (CIJob, error) {
+	var job CIJob
+	if err := node.Decode(&job); err != nil {
+		return job, err
+	}
+	job.FileReference = loadersUtils.GetFileReference(node)
+	return job, nil
+}
+
+func parseDeploymentJob(node *yaml.Node) (DeploymentJob, error) {
+	var job DeploymentJob
+	if err := node.Decode(&job); err != nil {
+		return job, err
+	}
+	job.FileReference = loadersUtils.GetFileReference(node)
+	return job, nil
+}
+
+func parseTemplateJob(node *yaml.Node) (TemplateJob, error) {
+	var job TemplateJob
+	if err := node.Decode(&job); err != nil {
+		return job, err
+	}
+	job.FileReference = loadersUtils.GetFileReference(node)
+	return job, nil
+}
+
+func getJobType(job *yaml.Node) JobType {
+	for _, node := range job.Content {
```

Comment:
Just so I understand, do you check every key of the object and check if it matches one of the existing job types?